migrator/replica/db_replica_manager_impl.go

```diff
@@ -0,0 +1,117 @@
+type DBReplicaManagerImpl struct {
+	forceRollbackVersion string
+	adminConfig          *pgxpool.Config
+	sourceMap            map[string]string
+	basePath             string
+	dbmRocks             *rocksdb.DBReplicaManagerImpl
+	dbmPostgres          *postgres.DBReplicaManagerImpl
+}
+
+// NewPostgres - returns a new ready-to-use store.
+func NewPostgres(basePath string, forceVersion string, adminConfig *pgxpool.Config, sourceMap map[string]string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		forceRollbackVersion: forceVersion,
+		adminConfig:          adminConfig,
+		sourceMap:            sourceMap,
+		basePath:             basePath,
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+		dbmPostgres:          postgres.New(forceVersion, adminConfig, sourceMap),
+	}
+}
+
+// New - returns a new ready-to-use store.
+func New(basePath string, forceVersion string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		forceRollbackVersion: forceVersion,
+		basePath:             basePath,
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+	}
+}
+
+// Scan - checks the persistent data of central and gather the replica information
+// from the supported databases.
+func (d *DBReplicaManagerImpl) Scan() error {
+	err := d.dbmRocks.Scan()
+	if err != nil {
+		// If our focus is Postgres, just log the error and ignore Rocks
+		if features.PostgresDatastore.Enabled() {
+			log.Warn(err)
+		} else {
+			return err
+		}
+	}
+
+	if features.PostgresDatastore.Enabled() {
+		err = d.dbmPostgres.Scan()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// GetReplicaToMigrate - finds a replica to migrate.
+// It returns the replica link, path to database, postgres database name and error if fails.
+func (d *DBReplicaManagerImpl) GetReplicaToMigrate() (string, string, string, error) {
+	// Get the RocksDB replica we are migrating
+	replica, replicaPath, err := d.dbmRocks.GetReplicaToMigrate()
+	if err != nil {
+		return "", "", "", err
+	}
+
+	if features.PostgresDatastore.Enabled() {
+		// Get the version of the Rocks Current so Postgres manager can use that info
+		// to determine what replica it needs to migrate.
+		var rocksVersion *migrations.MigrationVersion
+		// If we found no replica to migrate from Rocks no need to check its version.  That means the
+		// mount does not exist
+		if replica != "" && replicaPath != "" {
+			rocksVersion = d.dbmRocks.GetVersion(rocksdb.CurrentReplica)
+			rocksVersion.LastPersisted = d.dbmRocks.GetCurrentReplicaCreationTime()
+		}
+
+		pgReplica, migrateFromRocks, err := d.dbmPostgres.GetReplicaToMigrate(rocksVersion)
+		if err != nil {
+			return "", "", "", err
+		}
+
+		// If we need to migrate from rocks we need to return both
+		// rocks and postgres replicas.  Otherwise we just need one or
+		// the other.
+		if migrateFromRocks {
+			return replica, replicaPath, pgReplica, err
+		}
+		return "", "", pgReplica, err
```

Comment:
If it is not migrateFromRocks, since we called rocks replica manager before to select replica, we may have created a temp replica from the "current rocksdb"?

Reply:
yeah need to re-order that.  Since I use current to determine if we need to migrate, I simply need to process Postgres first here and only process rocks if we are in postgres mode and do need to migrate.  If we don't need to migrate, there is no reason to process Rocks.